yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -657,6 +657,9 @@ private[spark] class Client(
           case e: ApplicationNotFoundException =>
             logError(s"Application $appId not found.")
             return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
+          case ie: IOException =>
+            logError(s"Application $appId occurred an unexpected IOException.")
+            return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
```

Comment:
About the `run()` method, yeah, I missed there are other places that call this method.

But the exception comment still apply. There's no reason to cherry-pick exceptions here, just catch anything.